app/controllers/users/sessions_controller.rb

```diff
@@ -13,13 +13,21 @@ def new
 
   def create
     @theme = "minimal"
-
     respond_to do |format|
-      format.json {
-        self.resource = warden.authenticate!(scope: resource_name, recall: "#{controller_path}#new")
-        sign_in(resource_name, resource)
-        render status: 200, json: {status: "Success"}
-      }
+      format.json do
+        return auth_failed unless params[:user].present?
+
+        email = params[:user][:email]
+        password = params[:user][:password]
+        otp_attempt = params[:user][:otp_attempt]
```

Comment:
I think this might clean this up a bit:

```suggestion
        params[:user] => {email:, password:, opt_attempt:}
```

I think those are the same?

Reply:
Tried it but it doesn't work with params, It seems params doesn't implement the destructuring part of the hash interface. I found this though which works:

```ruby
email, password, otp_attempt = params[:user].values_at(:email, :password, :otp_attempt)
```

Using that in the newest revision. Happy to revert to how I had it before if you prefer though.